src/agent/debuglet.ts

```diff
@@ -115,6 +149,17 @@ export class Debuglet extends EventEmitter {
   private controller: Controller;
   private completedBreakpointMap: {[key: string]: boolean};
 
+  // breakpointFetchedTimestamp represents the last timestamp when
+  // breakpointFetched was resolved, which means breakpoint update was
+  // successful.
+  private breakpointFetchedTimestamp: number;
+  // breakpointFetched is a CachedPromise only to be resolved after breakpoint
+  // fetch was successful. Its stored promise will be returned by isReady().
+  private breakpointFetched: CachedPromise;
+  // debuggeeRegistered is a CachedPromise only to be resolved after debuggee
+  // registration was successful.
+  private debuggeeRegistered: CachedPromise;
```

Comment:
The only call to `debuggeeRegistered.refesh()` I see is in the constructor. This suggests to me that there is a single promise we use forever. This implies that `debuggeeRegistered.get().then(foo)` will run `foo` immediately (after a microtask hop).

This doesn't seem correct to me.

Reply:
Thanks for pointing out! This is a bug and I just have it fixed with the new commit. PTAL.